### PR TITLE
feat(curriculum): add english block 6.1 challenges

### PR DESCRIPTION
**Pull request by nieldakarla**
I've created the challenges for the block 6.1 of the English Curriculum. There are three dialogues and 61 tasks.

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-help-a-coworker-troubleshoot-on-github/656bc3bd0a323317d4117a49.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+# --description--
+
+The past continuous tense is used to talk about actions that were ongoing in the past. 
+
+It is formed using `was`  for the pronouns `I, he, she, it`), or `were` for the pronouns `you, we, they` followed by a verb ending in `-ing`. For example:
+
+`I was reading` means the work was happening over a period of time in the past. 
+
+PAst continuous is often used when an action is interrupted by another, as in `I was reading when the phone rang.` The reading was interupted by the phone ringing.
```
Comment (Sembauke) [inline curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-help-a-coworker-troubleshoot-on-github/656bc3bd0a323317d4117a49.md:16]:
```suggestion
Past continuous is often used when an action is interrupted by another, as in `I was reading when the phone rang.` The reading was interrupted by the phone ringing.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-help-a-coworker-troubleshoot-on-github/65a84bec88772eaff6e56679.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+dashedName: task-33
+---
+
+# --description--
+
+Imagine you're working on a project and have been trying different solutions over time. To express this ongoing effort, you cann use the present perfect tense. This tense connects past actions to the present moment. 
+
+The present perfect tense is formed using the auxiliary verb `have` or `has` followed by the past participle of the main verb. For example, `I have working on this for a long time` or `he has worked on this for a long time`.
+
+Let's practice using the present perfect tense to talk about actions that started in the past and are still relevant now.
```
Comment (Sembauke) [inline curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-help-a-coworker-troubleshoot-on-github/65a84bec88772eaff6e56679.md:14]:
```suggestion
Imagine you're working on a project and have been trying different solutions over time. To express this ongoing effort, you can use the present perfect tense. This tense connects past actions to the present moment.

The present perfect tense is formed using the auxiliary verb `have` or `has` followed by the past participle of the main verb. For example, `I have been working on this for a long time` or `he has worked on this for a long time`.

Let's practice using the present perfect tense to talk about actions that started in the past and are still relevant now.
```

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-help-a-coworker-troubleshoot-on-github/65a85418ea38cdc0a334dab2.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+# --description--
+
+`To Roll back` means going back to an earlier state or version of something. For example:
+
+`If the new app is not working, try rolling back to the old version.` 
+
+A `previous version` is an earlier form of something, like an older version of a software. Let's use it in the same sentence as above:
+
+`If the new app is not working, try rolling back to the previous version.`
```
Comment (Sembauke) [inline curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-help-a-coworker-troubleshoot-on-github/65a85418ea38cdc0a334dab2.md:22]:
```suggestion
`To roll back` means going back to an earlier state or version of something. For example:

`If the new app is not working, try rolling back to the old version.`

A `previous version` is an earlier form of something, like an older version of software. Let's use it in the same sentence as above:

`If the new app is not working, try rolling back to the previous version.`
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Review by DanielRosa74 — APPROVED** [review]
LGTM

File: curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-help-a-coworker-troubleshoot-on-github/65b2b0e08ec66535fa8542eb.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+---
+id: 65b2b0e08ec66535fa8542eb
+title: "Dialogue 3: Talking about Debugging"
```
Comment (moT01) [inline curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-help-a-coworker-troubleshoot-on-github/65b2b0e08ec66535fa8542eb.md:3]:
```suggestion
title: "Dialogue 3: Talking About Debugging"
```

**Review by moT01 — CHANGES_REQUESTED** [review]
A couple renaming suggestions to keep things consistent.

**Review by Sembauke — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Comment by miyaliu666** [comment]
@nieldakarla  Based on the order of dialogue, task-12 `Sarah: "Sure, Bob. What’s the problem?"` should be right after task-5 `Bob: Hey, Sarah. I was checking the branch we worked on yesterday when I saw a problem. I think we should open an issue on GitHub.` and before tasks 6-15.

And, I'm confused about tasks 6-15. Tasks 7-11 are teaching new words in `Bob: I pushed some changes to the repository earlier, but now the code won't compile. It's showing some strange errors.`, and tasks 13-15 are designed to practice these words -  I feel these should be before comprehension of this sentence, i.e. task 6.

**Pull request merged**